osu.Game/Screens/Edit/Compose/Components/Timeline/TimelineBlueprintContainer.cs

```diff
@@ -118,9 +120,44 @@ protected override void Update()
 
             base.Update();
 
+            updateSamplePointContractedState();
             updateStacking();
         }
 
+        public Bindable<bool> SamplePointContracted = new Bindable<bool>();
+
+        private void updateSamplePointContractedState()
+        {
+            // because only blueprints of objects which are alive (via pooling) are displayed in the timeline, it's feasible to do this every-update.
+
+            const double minimum_gap = 28;
+
+            // Find the smallest time gap between any two sample point pieces
+            double smallestTimeGap = double.PositiveInfinity;
+            double lastTime = double.PositiveInfinity;
+
+            // The blueprints are ordered in reverse chronological order
+            foreach (var selectionBlueprint in SelectionBlueprints)
```

Comment:
This can't be done. While `SelectionBlueprints` will only contain the visible blueprints in the general case... it will also contain more blueprints if they are selected. In fact if you select every object you may see these contract for no reason:

https://github.com/user-attachments/assets/5cff6f47-85d0-4e70-99fb-e0141c02de2f

I imagine this will also make selecting all objects even slower than it usually is, although crude profiling shows maybe not very much:

![1729505224](https://github.com/user-attachments/assets/7fda5bf6-1768-4f5f-9fd0-b8aa5481f4f9)

Reply:
You're very right. I forgot the Ctrl+A loads every single selection blueprint. I'll find some way to get only the visible blueprints.

The performance impact is probably small because it only does a simple operation per each blueprint.